### PR TITLE
Change tree colours when napari colourmap changed

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/lowe-lab-ucl/arboretum/issues/49

### CODE REVIEW

File: napari_arboretum/_tests/test_plugin.py [hunk 1]
```diff
@@ -1,17 +1,54 @@
...
-def test_plugin(make_napari_viewer):
-    """
-    A simple smoke test for drawing the graph. Note that this checks the code
-    works, not that the correct graph is drawn!
-    """
+@pytest.fixture
+def viewer_plugin(make_napari_viewer):
     viewer = make_napari_viewer()
     tracks, segmentation = load_sample_data()
+    viewer.add_layer(tracks)
```
[the remark this answers is not in the document]
Comment (dstansby) [reply napari_arboretum/_tests/test_plugin.py:18]: 👍 added

File: napari_arboretum/_tests/test_plugin.py [hunk 2]
```diff
@@ -1,17 +1,54 @@
...
+
+    old_color = new_color
+    # Change the color by attribute
+    assert viewer.layers[0].color_by != "generation"
+    viewer.layers[0].color_by = "generation"
+
+    # Check that color has changed
+    new_color = tree.get_branch_color(branch_id=id)
+    # Slice to remove alpha, which is 1 both before and after
+    assert np.all(new_color[:, :3] != old_color[:, :3])
```
[the remark this answers is not in the document]
Comment (dstansby) [reply napari_arboretum/_tests/test_plugin.py]: Good idea, I've split these

File: napari_arboretum/_tests/test_plugin.py [hunk 3]
```diff
@@ -1,17 +1,54 @@
...
+def test_plugin(viewer_plugin):
+    """
+    A simple smoke test for drawing the graph. Note that this checks the code
+    works, not that the correct graph is drawn!
+    """
+    viewer, plugin = viewer_plugin
     plugin.plotter.draw_tree(track_id=140)
+
+
+def test_color_changes(viewer_plugin):
```
[the remark this answers is not in the document]
Comment (dstansby) [reply napari_arboretum/_tests/test_plugin.py]: 👍 I've added docstrings for the new tests

**Review by alessandrofelder — APPROVED** [review]
🚀

**Pull request merged**